src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/GetWorkitemMetadata.sql

```diff
@@ -0,0 +1,54 @@
+﻿/*************************************************************
+    Stored procedure for getting a workitem metadata.
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     GetWorkitemMetadata
+--
+-- DESCRIPTION
+--     Gets a UPS-RS workitem metadata.
+--
+-- PARAMETERS
+--     @partitionKey
+--         * The system identifier of the data partition.
+--     @workitemUid
+--         * The workitem UID.
+--     @procedureStepStateTagPath
+--         * The Procedure Step State Tag Path.
+-- RETURN VALUE
+--     Recordset with the following columns
+--          * WorkitemUid
+--          * WorkitemKey
+--          * PartitionKey
+--          * Status
+--          * TransactionUid
+--          * ProcedureStepState Tag Value
+------------------------------------------------------------------------
+CREATE OR ALTER PROCEDURE dbo.GetWorkitemMetadata
+    @partitionKey                   INT,
+    @workitemUid                    VARCHAR(64),
+    @procedureStepStateTagPath      VARCHAR(64)
+AS
+BEGIN
+    SET NOCOUNT     ON
+    SET XACT_ABORT  ON
+
+    SELECT
+	    wi.WorkitemUid,
+	    wi.WorkitemKey,
+	    wi.PartitionKey,
+        wi.[Status],
+        wi.TransactionUid,
+	    eqt.TagValue AS ProcedureStepState
+    FROM 
+	    dbo.ExtendedQueryTagString eqt
+	    INNER JOIN dbo.WorkitemQueryTag wqt
+            ON wqt.TagKey = eqt.TagKey AND wqt.TagPath = @procedureStepStateTagPath
+	    INNER JOIN dbo.Workitem wi
```

Comment:
shouldn't we reverse the order of tables? We know workitem from partition and uid, and then can join smaller ranges of the other tables.

Reply:
Made some changes to address this. Please let me know your feedback.